### DISCUSSION TITLE
suppress cfgrib INFO when reading grib

### DISCUSSION THREAD

**Discussion by marstonsward** [pydata/xarray#7784] [Q&A] [answered]
The following produces a very annoying INFO from ECCODES when reading a grib file, `i`:
`grb = xr.open_dataset(i, engine='cfgrib')`
Results in several lines of the following:
`ECCODES INFO :  DjInDegrees is missing (See jDirectionIncrementGiven). Using value of 0.045011 obtained from La1, La2 and Nj`
How do I suppress this INFO? I can't seem to do it using JN warnings or display tricks.
Any tips?

**Comment by jhamman** [comment, marked as answer]
Not sure if this is something that can be controlled using Python's warnings library. You may find better help on the cfgrib repo: https://github.com/ecmwf/cfgrib

**Comment by douglatornell** [comment]
I have Python code that generates similar ECCODES messages @Marston, and I have not been able to suppress them. My guess is that the ECCODES C binary/library is writing them directly to stdout or stderr, but I haven't investigated deeply.

@jhamman is correct that `cfgrib` can be fairly noisy in addition to the ECCODES messages, but at least it can be controlled via the Python `warnings` and `logging` libraries.

**Reply by marstonsward** [reply]
@jhamman  & @douglatornell 

Thanks for the tips. It is annoying because it interferes with `TQDM` when looping over thousands of `grib` files. But I've not been able to control this info that either `xarray` passed to the JN or `JN` picks up everything from `stdout`. ECCODES is indeed a noisy package but I think I can live with this for now.
